### PR TITLE
support for django-rest-framework 3

### PR DESCRIPTION
**Pull request by craigds**
Builds on #14 and adds:

* Fixes the `serializer.bind()` issue mentioned at #14 (by binding serializers in `__init__`)
* tox.ini so you can easily run tests before pushing
* Removes support for old django versions; minimum is now 1.8. Minimum python version is now 2.7

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,9 +4,9 @@ python:
   - "3.3"
 env:
   - DJANGO_VERSION=1.4.16
-  - DJANGO_VERSION=1.5
   - DJANGO_VERSION=1.6
```
Comment (Ian-Foote) [inline .travis.yml]: You should drop 1.4 and 1.6 too.

File: .travis.yml [hunk 2]
```diff
@@ -4,9 +4,9 @@ python:
   - "3.3"
```
Comment (Ian-Foote) [inline .travis.yml]: This can be 2.7, 3.4 and 3.5

File: generic_relations/tests/models.py [hunk 3]
```diff
@@ -1,5 +1,11 @@
 from django.contrib.contenttypes.models import ContentType
-from django.contrib.contenttypes import generic
+
+try:
+    from django.contrib.contenttypes.fields import GenericForeignKey, GenericRelation
+except ImportError:
+    # django < 1.7
+    from django.contrib.contenttypes.fields import GenericForeignKey, GenericRelation
```
Comment (Ian-Foote) [inline generic_relations/tests/models.py]: This doesn't make sense.

Comment (craigds) [inline generic_relations/tests/models.py]: Ha, good catch. It was meant to be 
```
    from django.contrib.contenttypes.fields import GenericForeignKey, GenericRelation
```

However if we're explicitly dropping django 1.6 anyway there's no need for this

File: .travis.yml [hunk 4]
```diff
@@ -1,16 +1,17 @@
...
-  - DJANGO_VERSION=1.6
   - DJANGO_VERSION=1.7
+  - DJANGO_VERSION=1.8
+  - DJANGO_VERSION=1.9
 matrix:
     exclude:
-        - python: "3.3"
-          env: DJANGO_VERSION=1.4.16
+        - python: "3.5"
+          env: DJANGO_VERSION=1.7
```
Comment (Ian-Foote) [inline .travis.yml]: We might as well drop 1.7 too I think.

**Comment by craigds** [comment]
thanks @Ian-Foote :smile:

**Comment by Ian-Foote** [comment]
Thank you - I've been seriously neglecting this.

I've just released a `v1.0.0`.

**Pull request merged**